### PR TITLE
Administrator Guide/Upgrade Guide: add documentation for AstarteDefaultIngress

### PR DESCRIPTION
**Pull request by matt-mazzucato**
Add documentation for the new Managed Ingress (i.e.: AstarteDefaultIngress), and highlight the deprecation of AstarteVoyagerIngress.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,7 @@ and this project adheres to [Semantic Versioning](http://semver.org/spec/v2.0.0.
 ## [1.0.1] - Unreleased
 ### Added
 - [data_updater_plant] Add handle_data duration metric.
+- Add documentation for AstarteDefaultIngress.
```
Comment (Pavinati) [inline CHANGELOG.md:10]: We can also add something about voyager being now deprecated
```
- Add voyager deprecation notice.
```

File: doc/pages/administrator/062-handling_certificates.md [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+Let's Encrypt through cert-manager.
+
+The two alternative procedures for securing your Astarte deployment are outlined in the following
+sections.
+
+## Use your own certificates
+
+If you already own certificates for your domains, all it's needed is creating a TLS secret in the
+namespace in which Astarte resides. Assuming that the certificate and key are saved respectively as
+`cert.pem` and `privkey.pem`, simply run:
```
Comment (Pavinati) [inline doc/pages/administrator/062-handling_certificates.md:15]: when using voyager, `cert.pem` had to contain both the certificate and the CA bundle in a specific order. I think that's true even with the NGINX ingress. Should we add this to the doc?

File: doc/pages/administrator/062-handling_certificates.md [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+```bash
+$ kubectl create secret tls astarte-tls-cert -n astarte \
+  --cert=cert.pem --key=privkey.pem
+```
+
+## Use Let's Encrypt certificates with cert-manager
+
+The current section describes the procedure for setting up a `ClusterIssuer` to use Google CloudDNS
+to solve the DNS01 challenge. Therefore, when needed, the rest of this section will make use of the
```
Comment (Pavinati) [inline doc/pages/administrator/062-handling_certificates.md:25]: We should also describe how to setup the cluster issuer for HTTP challenges

File: doc/pages/administrator/062-handling_certificates.md [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+`cert.pem` and `privkey.pem`, simply run:
+
+```bash
+$ kubectl create secret tls astarte-tls-cert -n astarte \
+  --cert=cert.pem --key=privkey.pem
+```
+
+## Use Let's Encrypt certificates with cert-manager
+
+The current section describes the procedure for setting up a `ClusterIssuer` to use Google CloudDNS
```
Comment (Pavinati) [inline doc/pages/administrator/062-handling_certificates.md:24]: People reading this doc for the first time might not now what's a cluster issuer and how it relates to Astarte. I'd add a one-liner intro like
```
The process of obtaining a tls certificate from Let's Encrypt is handled by cert manager using a cluster issuer.
```
From there you can continue with something along the lines of
```
The issuer will query the Let's Encrypt API and handle the challenge to confirm you're the right owner of the specified domain.
We support two types of challenges, DNS and HTTP [...]
```

File: doc/pages/administrator/064-setup_astartedefaultingress.md [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+
+## API Paths
+
+`AstarteVoyagerIngress` deploys a well-known tree of APIs to the `host` you specified in the main `Astarte` resource.
+In particular, assuming your API host was `api.astarte.yourdomain.com`:
+
+* Housekeeping API base URL will be https://api.astarte.yourdomain.com/housekeeping/v1
+* Realm Management API base URL will be https://api.astarte.yourdomain.com/realmmanagement/v1
+* Pairing API base URL will be https://api.astarte.yourdomain.com/pairing/v1
+* AppEngine API base URL will be https://api.astarte.yourdomain.com/appengine/v1
```
Comment (Pavinati) [inline doc/pages/administrator/064-setup_astartedefaultingress.md:82]: base URLs don't contain the `/v1` portion

**Review by Pavinati — CHANGES_REQUESTED** [review]

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 6]
```diff
@@ -0,0 +1,211 @@
...
+    deploy: true
+```
+
+Further details on this topic can be found at [this page](064-setup_astartedefaultingress.html).
+
+When the `AstarteDefaultIngress` resource is created, the Operator is responsible for creating an
+NGINX ingress and a service of type LoadBalancer in the Astarte namespace.
+
+Wait until both the ingress and the new service have an assigned external IP, then update your Cloud
+DNS zone record sets with the new external IPs.
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:197]: IMO after this sentence we need to specify:
1. How to retrieve the external IPs and which IP to use for each DNS record
2. After setting the DNS records you have to be sure that the records are propagated, for devices as well. One way to do this is to wait for the records TTL before deleting the AVI.

File: doc/pages/administrator/064-setup_astartedefaultingress.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+  astarte: astarte
+  tlsSecret: <your-astarte-tls-cert>
+  api:
+    exposeHousekeeping: true
+  dashboard:
+    deploy: true
+    ssl: true
+    host: <your-astarte-dashboard-host>
+  broker:
+    deploy: true
```
Comment (davidebriani) [inline doc/pages/administrator/064-setup_astartedefaultingress.md:57]: A serviceType must be specified:

```yaml
  broker:
    deploy: true
    serviceType: LoadBalancer
```

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 8]
```diff
@@ -0,0 +1,211 @@
...
+  astarte: astarte
+  tlsSecret: <your-astarte-tls-cert>
+  api:
+    exposeHousekeeping: true
+  dashboard:
+    deploy: true
+    ssl: true
+    host: <your-astarte-dashboard-host>
+  broker:
+    deploy: true
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:188]: A serviceType must be specified:

```yaml
  broker:
    deploy: true
    serviceType: LoadBalancer
```

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 9]
```diff
@@ -0,0 +1,211 @@
...
+The dangling AstarteVoyagerIngress is not needed anymore. To delete the resource simply run:
+```bash
+$ kubectl delete avi -n astarte
+```
+
+at this point, to complete the procedure, the two dangling Voyager ingresses must be deleted:
+```bash
+$ kubectl delete ingresses.voyager.appscode.com -n astarte avi-api-ingress
+$ kubectl delete ingresses.voyager.appscode.com -n astarte avi-vernwmq-ingress
+ ```
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:211]: I got a bit confused here on the naming. Maybe it could help to specify the commands to find the dangling resources.

```
kubectl get astartevoyageringresses.api.astarte-platform.org -n astarte
```
and
```
kubectl get ingresses.voyager.appscode.com -n astarte
```

File: doc/pages/administrator/020-prerequisites.md [hunk 10]
```diff
@@ -12,18 +12,29 @@ In your local machine, you'll need two main tools:
...
-You don't need to create Voyager ingresses yourself - just the Operator itself is enough.
+You don't need to create NGINX ingresses yourself - just the Operator itself is enough.
+
+## Voyager (deprecated)
+
+Until Astarte v1.0.0, the only supported Managed Ingress was the
+[Voyager](https://github.com/appscode/voyager) based `AstarteVoyagerIngress`. Starting from Dec the
+31st 2021, according to the [Voyager
+announcement](https://blog.byte.builders/post/voyager-v2021.09.15/), the support for Voyager will be
+dropped as stated [here](https://github.com/astarte-platform/astarte/issues/613). An alternative
```
Comment (Annopaolo) [inline doc/pages/administrator/020-prerequisites.md:35]: We should add here some helm commands to install `ingress-nginx`, as we do for `cert-manager` below

File: doc/pages/administrator/062-handling_certificates.md [hunk 11]
```diff
@@ -0,0 +1,114 @@
...
+to solve the DNS01 challenge. Therefore, when needed, the rest of this section will make use of the
+`gcloud CLI`.
+
+If your Astarte deployment is hosted by another cloud provider, please refer to the cert-manager
+specific [docuementation](https://cert-manager.io/docs/configuration/acme/dns01/).
```
Comment (davidebriani) [inline doc/pages/administrator/062-handling_certificates.md:29]: typo in `docuementation`

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 12]
```diff
@@ -0,0 +1,211 @@
...
+```
+
+Render the Helm templates with the following:
+```bash
+helm template $ASTARTE_OP_RELEASE_NAME astarte/astarte-operator \
+    --namespace $ASTARTE_OP_RELEASE_NAMESPACE \
+    --output-dir $ASTARTE_OP_TEMPLATE_DIR
+```
+
+After these step you will find the updated CRDs within
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:57]: Typo `these step` -> `these steps`

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 13]
```diff
@@ -0,0 +1,211 @@
...
+the latest version will be fetched and used. If you choose to upgrade to a specific version of the
+chart by using the `--version` flag, please make sure to generate the updated CRDs template using
+the same chart version.
+
+By design, Astarte Operator's Helm charts cannot univocally be mapped to Operator's releases in a
+one-to-one relationship. However each chart is tied to a specific Operator's version, which is user
+configurable.
+
+Therefore, upgrading a chart lead to an Operator's upgrade if and only if the Operator's tag
+referenced by the chart is changed. You can check the Operator's tag binded to the chart simply
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:85]: Typo `binded to the chart` -> `bound to the chart`

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 14]
```diff
@@ -0,0 +1,211 @@
...
+```bash
+helm show values astarte/astarte-operator
+```
+
+As usual, you can use the usual `--version` flag to point to a specific chart version.
+
+## Upgrade Astarte
+
+To upgrade your Astarte instance simply edit the Astarte resource in the cluster updating the
+`version` field to the one you want to upgrate to.
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:97]: Typo `you want to upgrate to` -> `you want to upgrade to`

File: doc/pages/administrator/064-setup_astartedefaultingress.md (hunk 7 above)
Comment (Annopaolo) [inline doc/pages/administrator/064-setup_astartedefaultingress.md:57]: If one is to use a reserved/fixed loadBalancerIP, there might be some more work to do depending on the cloud provider.
GKE requires one to reserve(or promote) the external IP address (https://cloud.google.com/compute/docs/ip-addresses/reserve-static-external-ip-address#promote_ephemeral_ip); Azure does not.
Should we add a line on this?

Comment (matt-mazzucato) [reply doc/pages/administrator/064-setup_astartedefaultingress.md:57]: I wouldn't expose the detail in the doc at the moment. Anyway, details on this are present in the ADI crd [here](https://github.com/astarte-platform/astarte-kubernetes-operator/blob/release-1.0/apis/ingress/v1alpha1/astartedefaultingress_types.go#L87-L90). @bettio @rbino opinions on this?

File: doc/pages/administrator/050-handling_certificates.md [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+```
+
+#### Update the DNS Zone
+
+Once the external IP of the ingress controller is known, update your DNS Zone setting the retrieved
+IP for each of the following record sets:
+
+- `api.your-domain.example.com`
+- `dashboard.your-domain.example.com` (if deployed)
+- `broker.your-domain.example.com`
```
Comment (Pavinati) [inline doc/pages/administrator/050-handling_certificates.md:84]: You don't need a DNS Zone, you just require that the 3 domains all point to the right IP. For all we care they could be in 3 different domains handled by 3 different domain name providers.

For the sake of brevity i would merge `Define a DNS Zone for your project` and `Update the DNS Zone` into a single paragraph
```
#### Configure your DNS
Once the external IP of the ingress controller is known, make sure all your Astarte domains point to the NGINX Ingress controller IP

- `api.your-domain.example.com`
- `dashboard.your-domain.example.com` (if deployed)
- `broker.your-domain.example.com`
```

File: doc/pages/administrator/064-setup_astartedefaultingress.md [hunk 16]
```diff
@@ -0,0 +1,89 @@
+# Setting up Astarte Default Ingress
+
+Once your Cluster [is up and running](060-setup_cluster.html), to expose it to the outer world you
+need to set up an Ingress. Currently, the only managed and supported Ingress is based upon
+[NGINX](https://nginx.org/en/), and this guide will cover only this specific case.
+
+Please, note that the previously supported Voyager based ingress is deprecated and should not be
+used (see the [Voyager announcement](https://blog.byte.builders/post/voyager-v2021.09.15/) and the
+related Astarte [issue](https://github.com/astarte-platform/astarte/issues/613) for further
+details).
```
[the remark this answers is not in the document]
Comment (matt-mazzucato) [reply doc/pages/administrator/064-setup_astartedefaultingress.md:10]: Highlighted it is the only supported ingress also in the upgrade guide. I would keep it also here. Opinions?

File: doc/pages/upgrade/030-upgrade_100_101.md (hunk 13 above)
Comment (matt-mazzucato) [reply doc/pages/upgrade/030-upgrade_100_101.md]: That was bad! Thanks!

File: doc/pages/administrator/064-setup_astartedefaultingress.md [hunk 17]
```diff
@@ -0,0 +1,152 @@
...
+```bash
+apiVersion: ingress.astarte-platform.org/v1alpha1
+kind: AstarteDefaultIngress
+...
+spec:
+  ...
+  broker:
+    deploy: true
+    serviceType: LoadBalancer
+    loadBalancerIP: <same-IP-of-your-ingress>
```
Comment (Pavinati) [inline doc/pages/administrator/064-setup_astartedefaultingress.md:125]: I'm afraid that describing the `loadBalancerIP` after the copy-paste ADI CR above may lead people into skipping this important step.
We can still describe the use of that parameter in its own section, but i would add it to the standard CR so that people reading this doc for the first time cannot skip it

File: doc/pages/administrator/060-setup_cluster.md [hunk 18]
```diff
@@ -98,5 +78,13 @@ spec:
         memory: 6144M
 ```
 
-You can simply apply this resource in your Kubernetes cluster with `kubectl apply -f`. The Operator will take
-over from there.
+Starting from Astarte v1.0.1, traffic coming to the broker is TLS terminated ad VerneMQ level. The
+two fields controlling this features, namely `sslListener` and `sslListenerCertSecretName` can be
+found within the `vernemq` section of the Astarte CR. To provide more context on this topic:
```
Comment (Annopaolo) [inline doc/pages/administrator/060-setup_cluster.md:83]: I'd rephrase this:
"..In a nutshell, their meaning is:
- `sslListener` controls whether TLS termination is enabled at VerneMQ level or not, "

File: doc/pages/administrator/064-setup_astartedefaultingress.md [hunk 19]
```diff
@@ -0,0 +1,152 @@
+# Setting up Astarte Default Ingress
+
+Once your Cluster [is up and running](060-setup_cluster.html), to expose it to the outer world you
+need to set up an Ingress. Currently, the **only managed and supported Ingress*** is based upon
```
Comment (Annopaolo) [inline doc/pages/administrator/064-setup_astartedefaultingress.md:4]: typo: three `***`

**Review by Annopaolo — CHANGES_REQUESTED** [review]

File: doc/pages/administrator/064-setup_astartedefaultingress.md (hunk 16 above)
[the remark this answers is not in the document]
Comment (Annopaolo) [reply doc/pages/administrator/064-setup_astartedefaultingress.md:10]: Ok, it reads nicely also like this

File: doc/pages/administrator/064-setup_astartedefaultingress.md (hunk 7 above)
Comment (matt-mazzucato) [reply doc/pages/administrator/064-setup_astartedefaultingress.md:68]: Nevermind, added info about this.

File: doc/pages/administrator/060-setup_cluster.md (hunk 18 above)
Comment (matt-mazzucato) [reply doc/pages/administrator/060-setup_cluster.md]: thanks

File: doc/pages/administrator/064-setup_astartedefaultingress.md (hunk 17 above)
Comment (matt-mazzucato) [reply doc/pages/administrator/064-setup_astartedefaultingress.md:134]: Added the field with comments. Will it be enough? :cold_sweat:

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 20]
```diff
@@ -0,0 +1,236 @@
...
+As soon as both the ingress and the broker service have an assigned external IP, update your Cloud
+DNS zone record sets with the new external IPs. Assuming that your Astarte and AstarteDefaultIngress
+are named `astarte` and `adi` respectively and that they resides in the `astarte` namespace, you can
+find the external IPs with:
+
+```bash
+$ # retrieve information about the ingress
+$ kubectl get ingress -n astarte
+NAME              CLASS   HOSTS          ADDRESS   PORTS     AGE
+adi-api-ingress   nginx   <your-hosts>   X.Y.W.Z   80, 443   6s
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:218]: `adi-api-ingress` -> `adi-astarte-api-ingress`

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 21]
```diff
@@ -0,0 +1,236 @@
...
+```
+
+and
+
+```bash
+$ # retrieve information about the broker service
+$ kubectl get service -n astarte astarte-broker-service
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:225]: `kubectl get service -n astarte astarte-broker-service` -> `kubectl get service -n astarte adi-astarte-broker-service`

File: doc/pages/upgrade/030-upgrade_100_101.md [hunk 22]
```diff
@@ -0,0 +1,236 @@
...
+NAME                 TYPE           CLUSTER-IP     EXTERNAL-IP   PORT(S)         AGE
+adi-broker-service   LoadBalancer   x.x.x.x        A.B.C.D       443:32149/TCP   17s
```
Comment (davidebriani) [inline doc/pages/upgrade/030-upgrade_100_101.md:227]: `adi-broker-service` -> `adi-astarte-broker-service`

**Review by davidebriani — APPROVED** [review]

File: doc/pages/upgrade/030-upgrade_100_101.md (hunk 20 above)
Comment (matt-mazzucato) [reply doc/pages/upgrade/030-upgrade_100_101.md:218]: If the AstarteDefaultIngress is named `adi`, the ingress will be `adi-api-ingress` (see [here](https://github.com/astarte-platform/astarte-kubernetes-operator/blob/973884f968430dd75212622eec35bc65db61b07e/lib/defaultingress/api_ingress.go#L108-L110)).

**Review by bettio — APPROVED** [review]

**Comment by matt-mazzucato** [comment]
This PR falls in the context of #613.
